docs/build/custom.md

```diff
@@ -0,0 +1,184 @@
+* NOTE: Building will edit some of the ONNX Runtime source files to exclude unused kernels. If you wish to go back to creating a full build, or wish to change the operator kernels included, you MUST run `git reset --hard` or `git checkout HEAD -- ./onnxruntime/core/providers` from the root directory of your local ONNX Runtime repository to undo these changes.
+
+### Option to reduce types supported by the required operators
+
+**`--enable_reduced_operator_type_support`**
+
+* Enables [operator type reduction](../reference/ort-format-model-conversion.md#enable-type-reduction). Requires ONNX Runtime version 1.7 or higher and for type reduction to have been enabled during model conversion
+
+If the configuration file is created using ORT format models, the input/output types that individual operators require can be tracked if `--enable_type_reduction` is specified. This can be used to further reduce the build size if `--enable_reduced_operator_type_support` is specified when building ORT.
+
+ONNX format models are not guaranteed to include the required per-node type information, so cannot be used with this option.
+
+## Minimal build
+
+ONNX Runtime can be built to further minimize the binary size, by only including support for loading and executing models in [ORT format](../reference/ort-format-model-conversion.md), and not ONNX format.
+
+**`--minimal_build`**
+
+* RTTI is disabled by default in this build, unless the Python bindings (`--build_wheel`) are enabled.
+* If you wish to enable execution providers that compile kernels such as NNAPI or CoreML specify `--minimal_build extended`. See [here](../execution-providers/NNAPI-ExecutionProvider.md) and [here](../execution-providers/CoreML-ExecutionProvider.md) for details on using NNAPI and CoreML with ONNX Runtime Mobile
+
+A minimal build has the following limitations:
+
+* No support for ONNX format models, that is model must be converted to ORT format
+* No support for runtime optimizations. Optimizations are performed during conversion to ORT format
+* Limited support for runtime partitioning (assigning nodes in a model to specific execution providers). Execution providers that statically register kernels (e.g. ONNX Runtime CPU Execution Provider) are supported by default. All execution providers that will be used at runtime MUST be registered when creating the ORT format model
+    - Execution providers that compile nodes are optionally supported
+      - currently this is limited to the NNAPI and CoreML Execution Providers
+
+We do not currently offer backwards compatibility guarantees for ORT format models, as we will be expanding the capabilities in the short term and may need to update the internal format in an incompatible manner to accommodate these changes. You may need to regenerate the ORT format models to use with a future version of ONNX Runtime. Once the feature set stabilizes we will provide backwards compatibility guarantees.
```

Comment:
>We do not currently offer backwards compatibility guarantees for ORT format models, as we will be expanding the capabilities in the short term and may need to update the internal format in an incompatible manner to accommodate these changes. [](http://example.com/codeflow?start=0&length=241)

Can we update this to say that we now offer a backwards compatibility guarantee?

Reply:
Yes, if we do, that's great.

Reply:
Fixed